app/lib/features/calendar_sync/calendar_sync.dart

```diff
@@ -97,28 +96,37 @@ Future<void> initCalendarSync({bool ignoreRejection = false}) async {
         _log.info('ignoring state change without value');
         return;
       }
-      _scheduleRefresh(calendarId, events);
+      scheduleRefresh(calendarId, events);
     },
     fireImmediately: true,
   );
 }
 
+Completer<void>? _completer;
 Timer? _debounce;
 (String, List<EventAndRsvp>)? _next;
 bool _running = false;
 
 // schedules an update of the calender
 // makes sure there is only one running at a time
-void _scheduleRefresh(
+@visibleForTesting
+Future<void> scheduleRefresh(
   String calendarId,
   List<EventAndRsvp> events,
 ) {
   _debounce?.cancel(); // cancel the current debounce;
-  _debounce = Timer(const Duration(seconds: 3), () {
+  _completer ??= Completer<void>();
```

Comment:
this completer is reused which allows us to call scheduleRefresh multiple times receiving the same future to complete on the latest of its calls - which is used in the tests to ensure completion is done.